Version v11.2.0

danjm opened the pull request.

DDDDDanica committed:
fix(20316): enrich error message for fetch-with-cache (#20742)

albertolive committed:
[MMI] Added code fences to hide emojis just for MMI build (#20754)

* Added code fences to hide emojis just for MMI build

* Added missing code fences

* Added missing code fences

HowardBraham committed:
fix: "Allow quick Add Account name based on default name" (20168) (#20748)

I made a mistake in a suggestion on PR #20168, and Mark Stacey noticed it.  This PR fixes it.

georgewrmarshall committed:
Update account list to use all available screen space (#20745)

* Updating ModalContent styles and padding to allow for dynamic resizing of content that allows scrolling within the modal. This commit also updates the AccountListMenu so the account list can resize dynamically to take up all screen space available

* Updating the scroll bar styles from default for the ModalContent component so the scrollbar is thinner and more sleek

jpuri committed:
Updating ppom validator (#20756)

metamaskbot committed:
New Crowdin translations by Github Action (#20734)

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>

witmicko committed:
update codeql actions versions to non-deprecated (#20773)

darkwing committed:
Fix #682 - Show connected app favicon in bottom bar (#20636)

darkwing committed:
Fix #1203 - Add iframe for analytics during onboarding (#20693)

Gudahtt committed:
Add MMI builds to metamaskbot comment (#20783)

The `metamaskbot` comment on each PR now includes links to MMI builds.

This relates to https://github.com/MetaMask/metamask-extension/issues/20782

Gudahtt committed:
Publish MMI builds alongside stable releases (#20788)

MMI builds are now published alongside stable releases, in the same
manner as Flask. They will be included in future GitHub release pages,
and tracked using separate git tags. Sentry sourcemaps are uploaded for
them during the release process as well.

Closes https://github.com/MetaMask/metamask-extension/issues/20782

darkwing committed:
UX: Multichain: Remove hover background on Account Picker (#20794)

FrederikBolding committed:
Throw error if mock server port is busy (#20692)

* Throw error if mock server port is busy

* Fix lint

---------

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

zone-live committed:
[MMI] Shows Stake & Portfolio buttons and hides the Buy and Bridge buttons (#20767)

* shows MMI buttons and hides the Buy and Bridge buttons

* minor update

GuillaumeRx committed:
Fix broken `approveTokenDescriptionWithoutSymbol` translation key (#20807)

NidhiKJha committed:
Ux Multichain : Updated Select Action Modal (#20755)

* updated stake description

* updated stake copy

darkwing committed:
Multichain: Cleanup: Use constants, jest fns (#20678)

* Multichain: Cleanup: Use constants, jest fns

* Update ui/components/multichain/select-action-modal/select-action-modal.js

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

* Use BUYABLE_CHAIN_ETHEREUM_NETWORK_NAME

* Fix import issues

---------

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

OGPoyraz committed:
Remove `nonceDetails` from `TransactionMeta` (#20729)

darkwing committed:
UX: Fix errant global menu event (#20781)

georgewrmarshall committed:
Updating NetworkListMenu component to show as many networks in the network list as screen realestate allows (#20801)

jiexi committed:
Migrate MetaMaskController from mocha to jest (#20696)

* Converts MetaMaskController test from mocha to jest

Co-authored-by: Alex Donesky <adonesky@gmail.com>

---------

Co-authored-by: Alex Donesky <adonesky@gmail.com>

jpuri committed:
Adding env vars for blockaid to build file (#20769)

jpuri committed:
Revert ppom.js exactly yo audited version (#20826)

digiwand committed:
Fix: Change style object key format kebab -> camelCase and fix title param type (#20768)

* fix: style object key kebab -> camelCase

* fix: disable boolean (false) title in ItemList

NidhiKJha committed:
Migrate Import Token Link to TypeScript (#20474)

* updated ImportTokenLink component to use TS

* updated props

* added type file

* updated changes

---------

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

zone-live committed:
[MMI] Show the NFT tab content for mmi (#20830)

* allow NFTs content view

* prettier

* Update app/_locales/en/messages.json

Co-authored-by: Nidhi Kumari <nidhi.kumari@consensys.net>

---------

Co-authored-by: Nidhi Kumari <nidhi.kumari@consensys.net>

sahar-fehri committed:
fix: fix nft order (#20796)

* fix: fix nft order

* fix: use lodash

* fix: fix import

* Trigger Build

adonesky1 committed:
Integrate `SelectedNetworkController` (#20821)

* Import SelectedNetworkController, createSelectedNetwork middleware and integrate them such that wallet_switchEthereumChain API call the setNetworkClientIdForDomain method on the new controller

vinistevam committed:
fix: error during persisted transaction approval (#20626)

This pull request addresses the issue that occurs when a pending transaction is present, the extension is disabled and then re-enabled, and subsequently, the user rejects the transaction. The goal is to prevent the logging of the error message "User rejected the request", while still capturing and logging any other types of errors.

Fixes https://github.com/MetaMask/metamask-extension/issues/20604

FrederikBolding committed:
Only show enabled insight snaps on the confirmation screen (#20850)

FrederikBolding committed:
Use first non-symbol character as snap fallback icon (#20851)

* Skip first character in snap name if it is a @

* Simplify implementation

FrederikBolding committed:
Require Flask Snaps E2E to pass for CI pass (#20740)

FrederikBolding committed:
Fix spacing on connected sites in the snap details page (#20854)

dominikrudzki committed:
Make permissions modal scrollable (#20409)

FrederikBolding committed:
[FLASK] Bump snaps packages (#20567)

* Bump Snaps flask packages

* Add snap_getLocale support

* Update LavaMoat policies

* Bump iframe

* Update LavaMoat policies

* Bump iframe

* Update permission copy

* Add snap_getLocale description

* Fix unit test

* Move the `saveSnapKeyring` callback to the `SnapKeyring` constructor (#20786)

* Move the `saveSnapKeyring` callback to the `SnapKeyring` constructor

This commit removes `saveSnapKeyring` from the list of callbacks passed
to the Snap Controller since this function contains business logic
specific to the Snap Keyring.

Instead, `saveSnapKeyring` is injected into the Snap Keyring through a
new argument called `callbacks`.

In the future, the `callbacks` argument can be reused to pass other
callbacks.

* Update the Snap Keyring dependency

* chore: use release versions of `eth-snap-keyring` and `rpc-methods`

* chore: update `yarn.lock`

* chore: update `yarn.lock`

---------

Co-authored-by: Daniel Rocha <daniel@rocha.io>

digiwand committed:
build: add "test:unit:jest:watch" script to package.json to run jest tests on a node server (#20683)

digiwand committed:
Fix: Unable to edit token allowance / custom spending cap (#20804)

dhruvv173 committed:
Fix-BannerAlert to TS (#20655)

* BannerAlert to TS

* removing unnecessary code from README

dan437 committed:
Enable zkSync Era in Swaps (#20809)

gantunesr committed:
fix: add new account after an account management snap is re-installed (#20810)

* fix: remove SnapKeyring background reference

* fix: lint

georgewrmarshall committed:
Adding storybook files for the StepProgressBar (#20842)

pedronfigueiredo committed:
Update chromedriver version (#20818)

* update chromedriver version

* fix yarn.lock

albertolive committed:
[MMI] Added institutional selectors unit tests (#20833)

* Added institutional selectors unit tests

* Testing failure test

* Fixed test

* Fixed tests

* Fixed tests

albertolive committed:
Added Institution-background unit test (#20855)

albertolive committed:
[MMI] Changed the wrong privacy policy URL to the good one (#20884)

* Changed wrong url to good one

* Added CONSENSYS_PRIVACY_LINK

darkwing committed:
UX: Set initial background color to system theme (#20858)

GuillaumeRx committed:
Update `useScrollRequired` logic (#20889)

* Update `useScrollRequired` to only require to scroll to bottom one time

* remove console.log

* simplify logic

* add comment about magic number

matthewwalsh0 committed:
Fix undefined getCurrentNetworkTransactions on speed up (#20899)

danjm committed:
Merge remote-tracking branch 'origin/develop' into v11.2.0

danjm committed:
Version v11.2.0

zone-live committed:
[MMI] updates the "connect custodian account" flow (#20387)

* adds stories file and unit test

* adds the logic to open custodian url or not

* lint

* fixes styles, and clean up

* test update

* remove unused locale

* small update

* lint

* ran verify locales fix

* fix typo

* updates code with configuration api v2

* clean up

* lint and prettier

* test fix

* prettier

* packages update

* updates for config v2

* updates for config v2

* clean up

* Update LavaMoat policies

---------

Co-authored-by: Albert Olivé <albertolivecorbella@gmail.com>
Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>
Co-authored-by: Danica Shen <zhaodanica@gmail.com>

FrederikBolding committed:
Fix Solana derivation paths for snaps (#20968)

albertolive committed:
Add Custody component unit tests (#20907)

* Initial work

* updated tests

* Update test

* Finished adding tests

* prettier

* Fixed tests

* Changed custodian name to envName and fixed tests

* Updated tests

* Updated tests

zone-live committed:
[MMI] Updates custody-keyring package (#20982)

* updates custody-keyring package

* dedupe

chloeYue committed:
Stabilize e2e test (#20987)

zone-live committed:
[MMI] clean up MMI Portfolio Dashboard URLs (#20990)

* clean up urls

* test update

* lint

zone-live committed:
[MMI] fix property name and filter out custodian (#21004)

* prettier

* fix test

zone-live committed:
[MMI] adds Snaps support to MMI build (#20970)

## **Description**
We are enabling Snaps for MMI, by adding the necessary variables in
builds.yml.

## **Manual testing steps**

1. Go to https://snaps.metamask.io/
2. Connect a snap
3. Go back to the extension an check that the snap has been loaded

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>

chloeYue commented:
Testing scenario task list:

 [X] onboarding - create a wallet
 [X] onboarding - import a wallet
 [X] keyring - reset a wallet
 [X] transactions - cancel transaction
 [] transactions - send erc20 token origin MM
 [] transactions - send erc721 token origin MM
 [] transactions - send native token origin MM
 [X] transactions - speed up transaction
 [] tokens - import erc1155 token origin MM
 [X] tokens - import erc20 token origin MM
 [X] tokens - import erc721 token origin MM
 [X] ens - name resolution
 [X] phishing - warning page

chloeYue committed:
Merge branch 'master' into Version-v11.2.0

danjm committed:
Merge branch 'master' into Version-v11.2.0

danjm committed:
Ignored ripple-lib deprecation advisory, as we do not depend on it in MetaMask (#21182)

## **Description**
This PR addresses a `yarn audit` failure that we currently see on
develop. The failure is due to the deprecation of the `ripple-lib`
library used within `@trezor/blockchain-link@npm:2.1.8`. Although
imported in one of our dependencies, MetaMask does not depend on that
library, and it's code is not used by our codebase. We can safely ignore
this advisory.

## **Manual testing steps**

Not needed. Builds should pass without a `yarn audit` warning

danjm committed:
Merge remote-tracking branch 'origin/master' into Version-v11.2.0

danjm committed:
Update changelog for v11.2.0 (#21213)

danjm commented:
This received QA approval as of [c738239](https://github.com/MetaMask/metamask-extension/pull/20914/commits/c73823901b1000e1493ef3af2ee91ef89b00d864)

Since then, this PR has just merged in the changes of v11.1.1 and v11.1.2

Those changes affected NFT behaviour when different privacy toggles were off, and updated some MMI repos. I have QA'd the NFT behaviours, and done some QA on MMI, including Snaps on MMI. The QA passes, so I will merge.
